src/CaptureClient/GpuQueueSubmissionProcessor.cpp

```diff
@@ -412,7 +413,23 @@ std::vector<TimerInfo> GpuQueueSubmissionProcessor::ProcessGpuDebugMarkers(
       color->set_blue(static_cast<uint32_t>(completed_marker.color().blue() * 255.f));
       color->set_alpha(static_cast<uint32_t>(completed_marker.color().alpha() * 255.f));
     }
-    marker_timer.set_user_data_key(completed_marker.text_key());
+
+    const uint64_t text_key = completed_marker.text_key();
+    marker_timer.set_user_data_key(text_key);
+
+    // We have special handling for DXVK instrumentation that extracts the encoded group_id from
+    // the label's text. The encoding is:
+    // 'DXVK__vkFunctionName#GROUP_ID', where 'GROUP_ID' is the group id.
+    CHECK(string_intern_pool.contains(text_key));
+    const std::string& text = string_intern_pool.at(text_key);
```

Comment:
Would it make sense to pull this out as a testable function and fully unit test this piece of code?

Reply:
It seems right now we are only testing a single well-formed label.

Reply:
Makes sense. Will do.

Reply:
Done!